### PR TITLE
Machines deployments should run release commands before deployment

### PR DESCRIPTION
**Pull request by jsierles**

### CODE REVIEW

File: internal/command/deploy/machines.go [hunk 1]
```diff
@@ -85,41 +90,145 @@ func createMachinesRelease(ctx context.Context, config *app.Config, img *imgsrc.
...
 
+	machines, err := flapsClient.List(ctx, "")
+
+	if err != nil {
+		return
+	}
+
+	machines = helpers.Filter(machines, func(m *api.Machine) bool {
+		m, err = flapsClient.Get(ctx, m.ID)
+		fmt.Println(m.Config.Metadata)
```
Comment (dangra) [inline internal/command/deploy/machines.go:224]: remove before merging?

File: flaps/flaps.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func (f *Client) Start(ctx context.Context, machineID string) (*api.MachineStart
 	return out, nil
 }
 
-func (f *Client) Wait(ctx context.Context, machine *api.Machine) (err error) {
+func (f *Client) Wait(ctx context.Context, machine *api.Machine, state string) (err error) {
```
[the remark this answers is not in the document]
Comment (jsierles) [reply flaps/flaps.go:117]: Is there a place we use them already in flyd to rip from?

File: helpers/fp.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package helpers
+
+func Filter[T any](slice []T, f func(T) bool) []T {
```
[the remark this answers is not in the document]
Comment (jsierles) [reply helpers/fp.go:3]: Sounds good! I Just wanted to try this to see if it works.

File: internal/command/deploy/machines.go (hunk 1 above)
Comment (jsierles) [reply internal/command/deploy/machines.go:224]: Yup this is in flux still

File: helpers/fp.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dangra) [reply helpers/fp.go:3]: After that comment I saw you moved the function from already existent helpers package. I guess we can delay it to clean up in another PR anyways

File: flaps/flaps.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dangra) [reply flaps/flaps.go:117]: nope, I don't think so.

File: helpers/fp.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsierles) [reply helpers/fp.go:3]: Yeah I'll add in another PR if that's OK. There are a few other places we can clean up.

File: flaps/flaps.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsierles) [reply flaps/flaps.go:117]: I'll do this in a follow up PR too. Would like to think about how we can make this stuff reusable by other code that use the same states.

**Pull request merged**